accepted/future-releases/spread-collections/feature-specification.md

```diff
@@ -331,18 +331,28 @@ An expression like:
 ```
 
 Is syntactically parsed as `mapOrSetLiteral`. To determine whether it actually
-is a map or set, the surrounding context is used. Given an `mapOrSetLiteral`
-with context type C:
+is a map or set, the upwards and downwards type inference contexts are used.
 
-*   If `Set<Null>` is assignable to C, and `Map<Null, Null>` is not assignable
-    to C, then the collection is a set literal.
+Given a `mapOrSetLiteral`:
 
-*  Otherwise, it is a map literal.
+1.  If `Map<Null, Null>` is assignable to the literal's context type, then it
+    is a map literal.
+
+2.  Else, if `Set<Null>` is assignable to the literal's context type, then the
```

Comment:
Note that this is not the criteria that is specified for empty set literals, but there's active discussion about this here: https://github.com/dart-lang/sdk/issues/35431